### ISSUE TITLE
[Bug] USBC to HDMI Cosmo Support

### ISSUE THREAD

**Issue by zero77** [gemian/gemian#5]
At the moment usb c to hdmi dual screen support for the cosmo communicator is not working and the external display is not being detected at all when you connect a usb c to hdmi cable.

**Comment by adamboardman** [comment]
I've had a bash at this:
- [buster-external-rotate-fail](https://github.com/gemian/xf86-video-hwcomposer/tree/buster-external-rotate-fail) - xrandr support work to a degree crashes within glamor-hybris when hdmi is plugged in
- [buster-external-shadow-fail](https://github.com/gemian/xf86-video-hwcomposer/tree/buster-external-shadow-fail) - xrandr with shadow support, shows only the mouse pointer

I think investing a month in it is enough for now, time to work on something else and maybe come back to it later.

**Comment by zero77** [comment]
Thank you for having a look, are updates available via apt-get.

**Comment by adamboardman** [comment]
No those are source branches, as both attempts are entirely non-functional there is no point packing it up for anyone to try, its a build and fix it yourself situation. Anyone serious about trying to progress from where I've got to will also need to build the latest pushes to libhybris and kernel, these will make it through to apt once there is something else advantageous to push.

I did do some other experiments, one of which was to make the main pixmap a fixed size for dual screen layout but render it squished to the internal screen which proved that the xorg<->desktop link is messed up as even though the x position starts off to the right it somehow gets reset back to 0 so leaving the right side of the squished screen empty. So there are both problems with getting two screens worth of desktop into xorg as well as getting the external screen to give any output at all.
